nixos/modules/virtualisation/cri-o.nix

```diff
@@ -103,7 +103,10 @@ in
       cgroup_manager = "systemd"
       log_level = "${cfg.logLevel}"
       pinns_path = "${cfg.package}/bin/pinns"
-      hooks_dir = []
+      hooks_dir = [
```

Comment:
Maybe this instead:
```
hooks_dir = lib.optionals config.virtualisation.containers.ociSeccompBpfHook.enable [ ''"${config.boot.kernelPackages.oci-seccomp-bpf-hook}",'' ];
```

🤷‍♂️

Reply:
No, we want to unset (aka `[ ]`) it to avoid reading from pre-defined and non-existing dirs. :)

Reply:
Oh. So you do want `[ "" ]` if `config.virtualisation.containers.ociSeccompBpfHook.enable` evaluates to `false`?

Reply:
> Oh. So you do want `[ "" ]` if `config.virtualisation.containers.ociSeccompBpfHook.enable` evaluates to `false`?

No, it should be the same as before: `hooks_dir = []` :)

Reply:
Oh boy... I've been drinking too much [RFC42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md) kool aid lately... I thought this was a nix attribute set! Sorry for the noise.

On that note... this does seem like a perfect candidate to convert to using structured attributes via a `settings` option... :smile:

Reply:
No worries, happy to join the conversation! :)

Yeah maybe a re-structuring of the overall config would be a good follow-up.